File: 1150 - Ultrapassando Z/ultrapassando_z.cpp
#include <bits/stdc++.h>
using namespace std;
int main(){
    long long x, z;
    cin >> x >> z;
    while(z <= x){
        cin >> z;
    }
    long long soma = 0, atual = x;
    int qtd = 0;
    while(soma <= z){
        soma += atual;
        atual++;
        qtd++;
    }
    cout << qtd << endl;
}

File: 1150 - Ultrapassando Z/ultrapassando_z.py
import sys

dados = list(map(int, sys.stdin.read().split()))
idx = 0
x = dados[idx]; idx += 1
z = dados[idx]; idx += 1
while z <= x:
    z = dados[idx]; idx += 1
soma = 0
atual = x
qtd = 0
while soma <= z:
    soma += atual
    atual += 1
    qtd += 1
print(qtd)
